crates/wasi-common/src/snapshots/wasi_snapshot_preview1.rs

```diff
@@ -813,12 +813,15 @@ impl<'a> WasiSnapshotPreview1 for WasiCtx {
         Ok(nevents)
     }
 
+    // Not all implementations use this implementation.
+    #[allow(dead_code)]
     // This is just a temporary to ignore the warning which becomes a hard error
     // in the CI. Once we figure out non-returns in `wiggle`, this should be gone.
     #[allow(unreachable_code)]
     fn proc_exit(&self, rval: types::Exitcode) -> std::result::Result<(), ()> {
-        // TODO: Rather than call std::process::exit here, we should trigger a
-        // stack unwind similar to a trap.
+        // Note that this is just a default implementation. Host environments
+        // will often override this `proc_exit` implementation and do other
+        // things other than exit the host process.
```

Comment:
If both wasmtime and lucet override this implementation, maybe we could instead replace it with `unimplemented!("runtimes are expected to override this implementation")`?

Reply:
Done